### PR TITLE
Login

### PR DESCRIPTION
**Pull request by phamhuuluan**
==> Title: Login
==> Redmine ticket: https://edu-redmine.sun-asterisk.vn/issues/19841

==> Attachment:

### CODE REVIEW

**Comment by phamhuuluan** [comment]
ready

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  def new; end
+
+  def create
+    user = User.find_by email: params[:session][:email].downcase
+
+    if user&.authenticate params[:session][:password]
+      log_in user
+      redirect_to root_url
+    else
+      flash[:danger] = "Invalid email/password combination"
```
Comment (ngocvt-0484) [inline app/controllers/sessions_controller.rb]: i18n e ơi

File: app/models/user.rb [hunk 2]
```diff
@@ -16,6 +16,11 @@ class User < ApplicationRecord
   has_secure_password
   
   USER_PARAMS = %i(fullname email password password_confirmation).freeze
+ 
+  def User.digest(string)
```
Comment (ngocvt-0484) [inline app/models/user.rb]: bỏ ngoặc
dùng class < self e nha

File: app/views/layouts/_header.html.erb [hunk 3]
```diff
@@ -17,4 +13,23 @@
     </div>
     </div>
   <% end %>
+  <ul class="nav navbar-nav navbar-right">
+  <li><%= link_to t(".home"), root_path %></li>
```
Comment (ngocvt-0484) [inline app/views/layouts/_header.html.erb]: ident

**Comment by phamhuuluan** [comment]
ready

**Comment by phamhuuluan** [comment]
ready

**Pull request merged**